src/FS.Embedded/Manifest/ManifestDirectoryContents.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.Extensions.FileProviders.Embedded.Manifest
+{
+    internal class ManifestDirectoryContents : IDirectoryContents
+    {
+        private IFileInfo[] _entries;
+
+        public ManifestDirectoryContents(Assembly assembly, ManifestDirectory directory)
+        {
+            if (assembly == null)
+            {
+                throw new ArgumentNullException(nameof(assembly));
+            }
+
+            if (directory == null)
+            {
+                throw new ArgumentNullException(nameof(directory));
+            }
+
+            Assembly = assembly;
+            Directory = directory;
+        }
+
+        public bool Exists => true;
+
+        public Assembly Assembly { get; }
+
+        public ManifestDirectory Directory { get; }
+
+        public IEnumerator<IFileInfo> GetEnumerator()
+        {
+            return EnsureEntries().GetEnumerator();
+
+            IReadOnlyList<IFileInfo> EnsureEntries() => _entries = _entries ?? ResolveEntries().ToArray();
+
+            IEnumerable<IFileInfo> ResolveEntries()
+            {
+                if (Directory == ManifestEntry.UnknownPath)
+                {
+                    yield break;
+                }
+
+                foreach (var entry in Directory.Children)
+                {
+                    switch (entry)
+                    {
+                        case ManifestFile f:
+                            yield return new ManifestFileInfo(Assembly, f);
```

Comment:
Can we cache these? vanilla `EmbeddedFileProvider` did that I think and there's no reason we can't do the same here.

Reply:
I think a low level component like this shouldn't care about caching and let that for something higher level. I won't do anything right now as we don't have any data that suggest this would have a terrible impact on perf. If we discover that is causing trouble in the future, we can just cache it at the ManifestFileProvider level. (Note that there is some caching going on already on the enumeration)